### PR TITLE
WIP - Login with neomad API

### PR DESCRIPTION
**Pull request by fuentesloic**

### CODE REVIEW

**Comment by fuentesloic** [comment]
taking a real work flow from now with nice PR ever (as usual 👍)

File: components/Login/LoginForm.js [hunk 1]
```diff
@@ -15,15 +15,32 @@ export default class LoginForm extends Component {
...
+
+    fetch(url, {
+      method: 'POST',
+      body: JSON.stringify(data),
+      headers: {
+        Accept: 'application/json',
+        'Content-Type': 'application/json'
+      }
+    })
+    .then((response) => console.warn(response))
```
Comment (fuentesloic) [inline components/Login/LoginForm.js]: @vinyll gotta a response ... but it's the page document, no clue to understand why Neomad don't take datas in body

Comment (fuentesloic) [reply components/Login/LoginForm.js]: @vinyll same result with postman, so I guess I made a mistake in Neomad API ...

File: App.js [hunk 2]
```diff
@@ -17,10 +17,12 @@ export default class App extends Component {
       lat: null,
       long: null,
       userLogged: false,
-      communityMode: false
+      communityMode: false,
+      user: null
     }
 
     this._authorized = this._authorized.bind(this)
+    this._getUser = this._getUser.bind(this)
```
Comment (vinyll) [inline App.js]: I don't understand what these lines are for.
It seems like it doesn't do anything different than the object's method directly, does it?

File: App.js [hunk 3]
```diff
@@ -29,12 +31,19 @@ export default class App extends Component {
     })
   }
 
+  _getUser (user) {
+    this.setState({
```
Comment (vinyll) [inline App.js]: Why does a _get_ method does a _set_?

File: App.js [hunk 4]
```diff
@@ -29,12 +31,19 @@ export default class App extends Component {
...
+  }
+
   render () {
     return (
       <View style={styles.app}>
         <HeaderApp userLogged={this.state.userLogged} />
+        <Text>{this.state.username}</Text>
         <View>
-          <LoginForm />
+          <LoginForm _getUser={this._getUser} />
```
Comment (vinyll) [inline App.js]: Can't you just pass `user={ this.user }`?
The component logic seems broken to me. I'm no expert to React, but it reads weird that the get sets a state that actually is a json converter within the component

**Review by vinyll — COMMENTED** [review]
Not familiar with all this.
However the flow seems a little awkward to me.
I'd be curious to pair about this feature. 🤝

**Pull request closed without merging**